tests/toit/utils.toit

```diff
@@ -39,3 +52,56 @@ class ToitExecutable:
       full-command += ["--sdk-dir", sdk-dir_]
     full-command += args
     return pipe.run-program full-command
+
+  fork --with-test-sdk/bool=true args/List -> ForkResult:
+    full-command := [toit_run_, toit-bin-src_]
+    if with-test-sdk:
+      full-command += ["--sdk-dir", sdk-dir_]
+    full-command += args
+    fork-data := pipe.fork
+        true                // use_path
+        // We create a stdin pipe, so that qemu can't interfere with
+        // our terminal.
+        pipe.PIPE-CREATED   // stdin.
+        pipe.PIPE-CREATED   // stdout
+        pipe.PIPE-CREATED   // stderr
+        full-command.first
+        full-command
+    stdin := fork-data[0]
+    stdout := fork-data[1]
+    stderr := fork-data[2]
+    child-process := fork-data[3]
+
+    stdin.close
+
+    stdout-string-latch := monitor.Latch
+    stdout-task := task --background::
+      try:
+        bytes := #[]
+        catch --trace:
+          while chunk := stdout.read:
+            bytes += chunk
+          stdout-string-latch.set bytes.to-string
+      finally:
+        stdout.close
+
+    stderr-string-latch := monitor.Latch
+    stderr-task := task --background::
+      try:
+        bytes := #[]
+        catch --trace:
+          while chunk := stderr.read:
```

Comment:
Shouldn't you be using `stderr.in` and maybe `read-all`?

Reply:
done.